tools/eslint-rules/restrict-let-in-for-loops.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Prohibit the use of `let` in `for` loops.
+ * @author Jessica Quynh Tran
+ */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  create(context) {
+
+    const msg = 'Use of `let` declaration within a for-loop is ' +
+                'not recommended. Please use `var` instead.';
+
+    function testForLoop(node) {
+      return node.init && node.init.kind === 'let';
+    }
+
+    function testForInOfLoop(node) {
+      return node.left && node.left.kind === 'let';
+    }
+
+    /**
+     * Report function to test if the for-loop is declared using `let`.
+     */
+    function checkForLet(node) {
+      if (testForLoop(node) || testForInOfLoop(node)) {
+        context.report(node, msg);
```

Comment:
In my opinion, this ends up reporting the wrong column. It indicates the issue is in the column where `for` starts, but it should really report the column where `let` occurs. Is it possible to fix that? Maybe instead of returning a boolean, `testForLoop()` and friends can return the offending node and that can be passed to `context.report()` rather than the `for` node?